spmd/tensor/parallel/api.py

```diff
@@ -1,122 +1,195 @@
 # Copyright (c) Meta Platforms, Inc. and affiliates
 import torch
 import torch.nn as nn
-from typing import Sequence, Tuple
+from typing import Union, Dict
 from spmd.tensor import (
     distribute_module,
     distribute_tensor,
-    DTensor,
     Shard,
     Replicate,
     DeviceMesh,
-    Placement,
 )
 from spmd.tensor.parallel import TensorParallelMultiheadAttention
-from spmd.tensor.parallel.style import ParallelStyle, PairwiseParallel
+from spmd.tensor.parallel.style import PairwiseParallel, ParallelStyle
 from spmd.tensor.parallel.utils import _create_1d_device_mesh
 
 
-def replicate_input(
-    inputs: Sequence[torch.Tensor], device_mesh: DeviceMesh
-) -> Tuple[DTensor, ...]:
-    replicate = [Replicate()] * device_mesh.ndim
-    return tuple(
-        DTensor.from_local(tensor, device_mesh, replicate) for tensor in inputs
-    )
+__all__ = [
+    "parallelize_module",
+]
 
 
-def replicate_output(output: DTensor, device_mesh: DeviceMesh) -> torch.Tensor:
-    if isinstance(output, DTensor):
-        replicate = [Replicate()] * output.device_mesh.ndim
-        # TODO: can the output be left incontiguous?
-        return (
-            output.redistribute(output.device_mesh, replicate)
-            .to_local()
-            .contiguous()
-        )
+def parallelize_module(
```

Comment:
naming: Shall we call it `parallel_module`, or `parallelize_module`? which one do you think it's better?

Reply:
parallel to me is a noun, so I would still want to keep the verb `parallelize`. Is there any naming guidance for PyTorch?

Reply:
This make sense, let's keep it as `parallelize`, i don't think there's a naming guidance from pytorch, so we need to rationalize API by ourselves. Do you think we shall call `parallel_plan` as `parallel` is a noun/adj?

Reply:
I don't have a strong opinion on this one. Let me think more on this.